internal/pkg/service/common/configmap/visit.go

```diff
@@ -113,7 +113,7 @@ func doVisit(vc *VisitContext, cfg VisitConfig) error {
 			field.StructField = typ.Field(i)
 			field.OriginalPath = append(field.OriginalPath, vc.OriginalPath...)
 			field.OriginalPath = append(field.OriginalPath, orderedmap.MapStep(field.StructField.Name))
-			field.MappedPath = vc.MappedPath
+			field.MappedPath = append(field.MappedPath, vc.MappedPath...)
 			field.Type = field.StructField.Type
```

Comment:
:man_facepalming: it was failing in complex configuration use-cases, it is hard to cover it with a test (slice must be copied, cannot be shared between parent and child field definition).